src/java.desktop/share/classes/javax/swing/JScrollBar.java

```diff
@@ -753,11 +753,40 @@ public void stateChanged(ChangeEvent e)   {
         }
     }
 
-    // PENDING(hmuller) - the next three methods should be removed
+    /**
+     * Unlike most components, JScrollBar will derive the minimum size from
+     * the preferred size in one axis and a fixed minimum size in the other.
+     * Thus, it overrides {@code JComponent.setMinimumSize} contract
+     * that subsequent calls to getMinimumSize will return the
+     * same value as set in {@code JComponent.setMinimumSize}
+     *
+     * @param minimumSize the new minimum size of this component
+     */
+    public void setMinimumSize(Dimension minimumSize) {
+        super.setMinimumSize(minimumSize);
+    }
 
     /**
-     * The scrollbar is flexible along it's scrolling axis and
+     * Unlike most components, JScrollBar will derive the maximum size from
+     * the preferred size in one axis and a fixed maximum size in the other.
+     * Thus, it overrides {@code JComponent.setMaximumSize} contract
+     * that subsequent calls to getMaximumSize will return the
+     * same value as set in {@code JComponent.setMaximumSize}
+     *
+     * @param maximumSize the desired maximum allowable size
+     */
+    public void setMaximumSize(Dimension maximumSize) {
+        super.setMaximumSize(maximumSize);
+    }
+
+    /**
+     * The scrollbar is flexible along its scrolling axis and
      * rigid along the other axis.
+     * As specified in {@code setMinimumSize} JScrollBar will derive the
+     * minimum size from the preferred size in one axis and a
+     * fixed minimum size in the other.
+     *
+     * @return the minimum size as specified above.
```

Comment:
The first sentence of the javadoc is put into the method summary, therefore the first sentence should explain what is returned. Is it worth addressing it too?

Something like: Returns the minimum size for the {@code JScrollBar}. The minimum size is equal to a fixed minimum value along its scrolling axis and the preferred size along the other axis.